### PR TITLE
[Refactor](mutex) Use std::mutex to replace Mutex and refactor some lock logic

### PR DESCRIPTION
**Pull request by caiconghui**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/tablet.h [hunk 1]
```diff
@@ -143,21 +143,9 @@ class Tablet : public BaseTablet {
...
-    inline void obtain_base_compaction_lock() { _base_lock.lock(); }
-    inline void release_base_compaction_lock() { _base_lock.unlock(); }
-    inline Mutex* get_base_lock() { return &_base_lock; }
-
-    // cumulative lock
-    inline void obtain_cumulative_lock() { _cumulative_lock.lock(); }
-    inline void release_cumulative_lock() { _cumulative_lock.unlock(); }
-    inline Mutex* get_cumulative_lock() { return &_cumulative_lock; }
+    inline std::mutex& get_push_lock() { return _ingest_lock; }
+    inline std::mutex& get_base_lock() { return _base_lock; }
```
Comment (yiguolei) [inline be/src/olap/tablet.h:147]: maybe get_base_compaction_lock better? and get_cumulative_compaction_lock()?

Comment (yiguolei) [reply be/src/olap/tablet.h:147]: And I think we could use only one lock... just use get_compaction_lock()?

**Comment by yiguolei** [comment]
LGTM

**Comment by platoneko** [comment]
Since C++17, std::unique_lock, std::lock_guard can omit template arguments. see https://en.cppreference.com/w/cpp/language/class_template_argument_deduction

**Review by weizuo93 — APPROVED** [review]
LGTM

**Pull request merged**